tests/test_abstract_repr.py

```diff
@@ -89,6 +89,43 @@
 )
 
 
+@pytest.mark.parametrize(
+    "test_torch",
+    [True, False],
+)
+def test_abstract_repr_encoder(test_torch: bool):
+    encoder = AbstractReprEncoder()
+
+    if not test_torch:
```

Comment:
You can avoid having everything in an if-else block by starting with 
```
if test_torch:
    torch = pytest.importorskip("torch")
```

Reply:
I don't see what you mean here. The way I set it up, it will test the non-torch part in the one path, and the torch path, if available, in the other. what you suggests seems to be an all-or-nothing approach. Am I missing something?

Reply:
What I suggest is that you do the tests that do not require `torch` regardless